### ISSUE TITLE
Unet-resnet50 - trained on S1S2water dataset - fails for some thumbnails of the same image (but performs correctly for others)

### ISSUE THREAD

**Issue by laetitialalla** [MWieland/s1s2_water#14]
Hello, 

I have an issue, and I would like to share it with the rest of the community to see if others have had the same problem before. 
- I trained a Unet (encoder ResNet50) with the S1S2water dataset to detect water on S1 radar images only (no S2 optic, no slope). 
- I want to use it to follow a specific lake in time : at some dates it works perfectly, but at other dates it fails. 
Like in the examples below : 20240311 and 20240323

<img width="2344" height="1062" alt="Image" src="https://github.com/user-attachments/assets/c6d4ee7e-b733-4305-a8aa-32cb5829728f" />

- Since the lake is quite big, I split the radar images into 256x256pixels thumbnails. And, within the same day, it can work perfectly on one thumbnail and fail on the other. We can actually visualise the bad result for the thumbnail at the bottom, but not for the thumbnail at the top-left. 

<img width="1035" height="523" alt="Image" src="https://github.com/user-attachments/assets/0881f7de-5afc-41c5-bddf-53d118dec871" />

- The histograms of these 2 thumbnails look very similar : 

<img width="2779" height="2382" alt="Image" src="https://github.com/user-attachments/assets/462c45a6-2760-4d56-9e5e-34bdde53669d" />

So it cannnot be : 
- a lack of examples in the training data : since it works perfectly for some dates. 
- an issue of the input image (weather, etc) : since some parts of the lake are correctly detected (all the input images are preprocessed in the same way with the same software). 

I even tried different loss functions, but I still have the same issue. 

Honestly, I am a bit puzzled. I have no idea why it does not work.  

@MWieland have you encoutered something like this when you trained your networks ? When you tried to predict an entire sentinel tile, it was not working on some patches ? 

Hope you can help !

**Comment by MWieland** [comment]
Dear @laetitialalla this looks rather unusual, especially given that this is a timeseries over the same spatial extent. We have had issues with our own models that showed similar "tiling artifacts", but these were real edge cases that weren't properly represented in the training dataset. I would rule this out here. 

After talking to a colleague of mine today about this issue, he suggested that this could be related to improper nodata handling. Even if a single pixel is assigned an out of range nodata value (e.g. -9999, which is often used by rasterio), it could mess up the inference. Depending on how you computed the histograms they may actually ignore nodata values and make it look alright.

Other than that you could try to play around with your tiling scheme (add more overlap between tiles, change tile size, etc.) and see if this can improve the results. It would, however, likely not solve the actual root cause of this issue.

**Comment by laetitialalla** [comment]
Dear @MWieland , thank you so much for helping me with this topic ! 

It is indeed unsual, hence my confusion xD 

Thank you for suggesting the hypothesis of the no-data wrong handling. I looked into it, but unfortunatly, it's not it :/ 
I looked at both images which were fed to the network and plotted the pixels out of the histogram values (to see if I missed something), but no... There are no nan values, neither -9999 values.

This is the tile that is not detected well : 
There are only 5 pixels that are not between the range [-1.5, 1.5] (in red on the image but it's realy tiny) and that are have finite values close to 1.5... 
<img width="2779" height="2385" alt="Image" src="https://github.com/user-attachments/assets/07fa1f11-d387-4e78-b625-ceb2b414b63f" />

This is the tile that is correctly detected : 
<img width="2779" height="2385" alt="Image" src="https://github.com/user-attachments/assets/1f21d705-8e28-4040-a44d-16288f3f39fe" />

Is that what you meant ? 

Worse, on the different versions of the network i trained (V0 / V1 / V2 : same architecture but different loss functions), it is not the same tile that is wrong... But the tile at the top-left is always correct... 

<img width="5573" height="2382" alt="Image" src="https://github.com/user-attachments/assets/6a505e7c-20a0-4caf-9434-3fd81859a8bf" />

I will play around with the tiling scheme to see if it fixes the problem (I could also mark these as outliers in my time serie). 
But I would really like to know why the network is not working everywhere ^^ This is not a usual behavior I guess... 

What kind of "tiling artefact" you had in your model ?

**Comment by MWieland** [comment]
Dear @laetitialalla apologies for the late response. Is the issue still persisting or could you solve it? From your response it looks as if we can rule out nodata as cause. I would, however, still suggest to try and add a "fillnodata" method (e.g. [gdal_fillnodata](https://gdal.org/en/stable/programs/gdal_fillnodata.html)) prior to normalization and prediction steps - just to be sure. 

Other than that, I was wondering about your normalization scheme. Can you confirm how normalization is applied at inference - are you using fixed mean/std (or min/max) values computed once from the training split and apply it to every new scene, or are the statistics recomputed per-scene/per-tile from the new image itself? If it's the latter, that could potentially also explain this effect. The same backscatter value could get mapped to different normalized values depending on what is in the rest of that particular scene/tile, even though the raw water values themselves look consistent across your time series.
